lib/Core/Executor.cpp

```diff
@@ -3706,14 +3706,21 @@ void Executor::executeMemoryOperation(ExecutionState &state, bool isWrite,
     ref<Expr> offset = mo->getOffsetExpr(address);
     ref<Expr> boundsCheck = mo->getBoundsCheckOffset(offset, bytes);
 
+
     bool inBounds;
-    solver->setTimeout(coreSolverTimeout);
-    bool success = solver->mustBeTrue(state, boundsCheck, inBounds);
-    solver->setTimeout(0);
-    if (!success) {
-      state.pc = state.prevPC;
-      terminateStateEarly(state, "Query timed out (bounds check).");
-      return;
+    if(NoBoundCheck){
```

Comment:
Also need to use `#ifdef ENABLE_Z3` to enclose the usage of `NoBoundCheck` which is only defined when `ENABLE_Z3` is defined.